pds_doi_core/cmd/pds_doi_cmd.py

```diff
@@ -0,0 +1,242 @@
+    m_doiPDS4LabelUtil = DOIPDS4LabelUtil()
+    m_doiValidatorUtil = DOIValidatorUtil()
+    m_doiWebClient = DOIWebClient()
+
+    def reserve_doi_label(self, target_url, publisher_value, contributor_value):
+        """
+        Function receives a URI containing either XML, SXLS or CSV and create one or many labels to disk and submit these label(s) to OSTI.
+        :param target_url:
+        :param publisher_value:
+        :param contributor_value:
+        :return:
+        """
+
+        action_type = 'reserve_osti_label'
+        o_doi_label = 'invalid action type:action_type ' + action_type
+
+        logger.debug(f"target_url,action_type {target_url} {action_type}")
+
+        file_is_parsed_flag = False
+
+        if target_url.endswith('.xml'):
+            o_doi_label = self.m_doiPDS4LabelUtil.parse_pds4_label_via_uri(target_url, publisher_value, contributor_value)
+
+            file_is_parsed_flag = True
+
+        if target_url.endswith('.xlsx'):
+            xls_filepath = target_url
+            # Get the default configuration from external file.  Location may have to be absolute.
+            xml_config_file = os.path.join('.','config','default_config.xml')
+
+            logger.debug(f"xml_config_file {xml_config_file}")
+            (dict_configList, dict_fixedList) = self.m_doiConfigUtil.get_config_file_metadata(xml_config_file)
+
+            app_base_path = os.path.abspath(os.path.curdir)
+
+            dict_condition_data = {}
+
+            (o_num_files_created,
+             o_aggregated_DOI_content) = self.m_doiInputUtil.parse_sxls_file(app_base_path,
+                                                                           xls_filepath,
+                                                                           dict_fixedList=dict_fixedList,
+                                                                           dict_configList=dict_configList,
+                                                                           dict_ConditionData=dict_condition_data)
+            o_doi_label = o_aggregated_DOI_content
+            file_is_parsed_flag = True
+            logger.debug(f"o_num_files_created {o_num_files_created}")
+            logger.debug(f"o_aggregated_DOI_content {o_aggregated_DOI_content}")
+
+        if target_url.endswith('.csv'):
+            xls_filepath = target_url
+            # Get the default configuration from external file.  Location may have to be absolute.
+            xml_config_file = os.path.join('.','config','default_config.xml')
+
+            (dict_configList, dict_fixedList) = self.m_doiConfigUtil.get_config_file_metadata(xml_config_file)
+
+            app_base_path = os.path.abspath(os.path.curdir)
+
+            dict_condition_data = {}
+
+            (o_num_files_created,
+             o_aggregated_DOI_content) = self.m_doiInputUtil.parse_csv_file(app_base_path,
+                                                                          xls_filepath,
```

Comment:
Ideally we would like to manage InputFormatException that will be raised by the input function. The exception will be catch and then the error can be logger and exit with code 0. The Exception management will be useful when we will integrate the code in the API. This is also the most generic way to manage this error cases compared to specific codes as done in o_doi_label.startswith('invalid').

Reply:
Exception handling has been added when calling services from input_util.py

Reply:
The returned parameter is now a dict.